apps/enc.c

```diff
@@ -268,9 +278,11 @@ int enc_main(int argc, char **argv)
         goto opthelp;
     }
 
-    if (cipher && EVP_CIPHER_flags(cipher) & EVP_CIPH_FLAG_AEAD_CIPHER) {
-        BIO_printf(bio_err, "%s: AEAD ciphers not supported\n", prog);
-        goto end;
+    if (cipher && (EVP_CIPHER_flags(cipher) & EVP_CIPH_FLAG_AEAD_CIPHER)) {
+        if ((EVP_CIPHER_mode(cipher) != EVP_CIPH_GCM_MODE)) {
```

Comment:
Why are you restricting this to GCM only? It seems that it would not be too difficult to make this generic.

Reply:
I'm with @mattcaswell here, why not CCM?

Reply:
CCM tests fail in my alternative implementation #5026, maybe that's the reason? Other ciphers work fine.

Reply:
CCM doesn't support streaming https://github.com/openssl/openssl/pull/5026#issuecomment-355779385